External/Plugins/LintingHelper/Managers/LintingManager.cs

```diff
@@ -0,0 +1,200 @@
+
+                list.Add(file);
+            }
+            
+            foreach (var lang in filesByLang.Keys)
+            {
+                LintFiles(filesByLang[lang].ToArray(), lang);
+            }
+        }
+
+        public static void LintDocument(ITabbedDocument doc)
+        {
+            var files = new string[] { doc.FileName };
+            var language = doc.SciControl.ConfigurationLanguage;
+
+            LintFiles(files, language);
+        }
+
+        /// <summary>
+        /// Runs all applicable linters on the current document
+        /// </summary>
+        public static void LintCurrentDocument()
+        {
+            LintDocument(PluginBase.MainForm.CurrentDocument);
+        }
+
+        /// <summary>
+        /// Applies the results to all open files
+        /// </summary>
+        /// <param name="results"></param>
+        static void ApplyLint(string[] files, List<LintingResult> results)
+        {
+            Cache.Clear(PluginBase.MainForm.Documents.Select( (d) => d.FileName ));
+
+            PluginBase.MainForm.CallCommand("PluginCommand", "ResultsPanel.ClearResults");
+
+            foreach (var file in files)
+            {
+                Cache.RemoveDocument(file);
+                DocumentManager.FindDocument(file)?.SciControl.RemoveHighlights();
+            }
+
+            if (results == null)
+            {
+                return;
+            }
+
+            Cache.AddResults(results);
+
+            foreach (var result in results)
+            {
+                var line = result.File + ":" + result.Line + ": " + result.Severity.ToString() + ": " + result.Description;
+
+                int state = 0;
+                if (result.Severity == LintingSeverity.Warning)
+                {
+                    state = -3;
+                }
+                TraceManager.Add(new TraceItem(line, state));
+                
+                var doc = DocumentManager.FindDocument(result.File);
+                if (doc != null)
+                {
+                    var start = doc.SciControl.PositionFromLine(result.Line - 1);
+                    var len = doc.SciControl.LineLength(result.Line - 1);
+                    start += result.FirstChar;
+                    if (result.Length > 0)
+                    {
+                        len = result.Length;
+                    }
+                    else
+                    {
+                        len -= result.FirstChar;
+                    }
+
+                    var color = 0;
+                    switch (result.Severity)
+                    {
+                        case LintingSeverity.Error:
+                            color = 0x00000080;
+                            break;
+                        case LintingSeverity.Warning:
+                            color = 0x00008080;
+                            break;
+                        case LintingSeverity.Info:
+                            color = 0x00008000;
```

Comment:
@Meychi Do we already have colors that we can reuse here? if not, which ones should we use for blending and/or deriving proper ones?